image/image.go

```diff
@@ -28,3 +37,44 @@ func ValidateDestinationTags(daemon bool, repoNames ...string) error {
 
 	return nil
 }
+
+func VerifyRegistryAccess(regInputs RegistryInputs, keychain authn.Keychain) error {
+	if len(regInputs.ReadableImages()) > 0 {
+		for _, imageRef := range regInputs.ReadableImages() {
+			err := verifyReadAccess(imageRef, keychain)
+			if err != nil {
+				return err
+			}
+		}
+	}
+
+	if len(regInputs.WriteableImages()) > 0 {
+		for _, imageRef := range regInputs.WriteableImages() {
+			err := verifyReadWriteAccess(imageRef, keychain)
+			if err != nil {
+				return err
+			}
+		}
+	}
+	return nil
+}
+
+func verifyReadAccess(imageRef string, keychain authn.Keychain) error {
+	if imageRef != "" {
+		img, _ := remote.NewImage(imageRef, keychain)
+		if !img.CheckReadAccess() {
+			return errors.New(fmt.Sprintf("read image %s from the registry", imageRef))
```

Comment:
```suggestion
			return errors.New(fmt.Sprintf("ensure read access to %s", imageRef))
```
(nit: this seems closer to what we're actually doing...)

Reply:
Fixed in the following [commit](https://github.com/buildpacks/lifecycle/pull/646/commits/5c900400a2e89ae0d9f8a8d0c02243dec574ceba)